python/pyspark/sql/dataframe.py

```diff
@@ -3912,16 +3912,29 @@ def union(self, other: "DataFrame") -> "DataFrame":
         >>> df2 = spark.createDataFrame([(3, "Charlie"), (4, "Dave")], ["id", "name"])
         >>> df1 = df1.withColumn("age", lit(30))
         >>> df2 = df2.withColumn("age", lit(40))
-        >>> df3 = df1.union(df2)
-        >>> df3.show()
-        +-----+-------+---+
-        | name|     id|age|
-        +-----+-------+---+
-        |Alice|      1| 30|
-        |  Bob|      2| 30|
-        |    3|Charlie| 40|
-        |    4|   Dave| 40|
-        +-----+-------+---+
+        >>> if spark.conf.get("spark.sql.ansi.enabled") == "false":
```

Comment:
Can we fix the input types, for example, floats and ints so it can pass whether `spark.sql.ansi.enabled` is on or off?